<?xml version="1.0"?>
<launch>

  <!-- ALL ROBOT SPECIFIC TASKS -->

  <group ns="robot2">
    <param name="tf_prefix" value="robot2" />
    <arg name="tf_prefix" value="robot2" />

  <node pkg="kmo_navserver" type="kmo_navserver_node" name="kmo_navserver" output="screen">
     <param name="host" value="192.168.100.100" />
     <param name="enc" value="true" />
     <param name="state" value="true" />
     <param name="laser" value="true" />
     <param name="sick" value="true" />
     <param name="steering_fix_wheel_distance" value="1.190" />
     <param name="steering_fix_wheel_distance_y" value="0.0" />
     <param name="odom_frame_id" value="$(arg tf_prefix)/odom" />
     <param name="base_footprint_frame_id" value="$(arg tf_prefix)/base_footprint" />
     <param name="state_frame_id" value="$(arg tf_prefix)/base_footprint_ground_truth" />
     
     
     <param name="tf_x" value="1.5" />
     <param name="tf_y" value="0.0" />
     <param name="tf_z" value="0.05" />
     <remap from="vmc_navserver/laserscan" to="laserscan"/>

     <param name="use_vmc_timestamp" value="false"/>
  </node>

  <node pkg="kmo_fork_control" type="fork_control_cititruck_node" name="fork_control" output="screen">
     <param name="visualize" value="true" />
     <param name="fork_max_moving_time" value="2.5" />
     <param name="fork_moving_time_support_to_low" value="0.3" />
     <param name="delay_support_to_low" value="0.2" />
     <remap from="/$(arg tf_prefix)/control/fork_report" to="/$(arg tf_prefix)/control/fork/report"/>
  </node>

  <node pkg="orunav_mpc" type="controller_node" name="controller" output="screen">
    <param name="robot_id" value="2" />
    <remap from="/$(arg tf_prefix)/cmd_vel"
	   to="/$(arg tf_prefix)/controller/cmd_vel" />
    <remap from="controller/commands"
	   to="control/controller/commands" />
    <remap from="controller/reports"
	   to="control/controller/reports" />
    <remap from="controller/trajectories"
	   to="control/controller/trajectories" />
  </node>

  <node pkg="orunav_motion_planner" type="get_path_service" name="get_path_service" output="screen">		
    <param name="model" value="CiTiTruck_Expanded10+20_16_1_4.0_0.2.reduced"/>
    <param name="motion_primitives_directory" value="$(find orunav_motion_planner)/Primitives/" />
    <param name="lookup_tables_directory" value="$(find orunav_motion_planner)/LookupTables/" /> 
    <param name="min_incr_path_dist" value="0.2" />
    <param name="visualize" value="true" />
  </node>

  <node pkg="orunav_constraint_extract" type="polygonconstraint_service" name="polygonconstraint_service" output="screen">
    <param name="load_type" value="1" />
    <param name="model_type" value="2" />
    <param name="visualize" value="false" />
    <param name="save_constraints_and_path" value="true" />
    <param name="lookuptables_file" value="$(find orunav_constraint_extract)/tables/service_lookup_model2load1.dat" />
    <param name="visualize_outer_constraints" value="false" />
    <param name="visualize_only_invalid" value="false" />
    <param name="skip_overlap" value="true" />
    <param name="save_lookuptables" value="false" />
  </node>  

  <node pkg="orunav_path_smoother" type="smoothed_path_service" name="smoothed_path_service" output="screen">
    <param name="visualize" value="true" />
    <param name="visualize_deep" value="false" />
    <param name="wheel_base" value="1.190" />
    <param name="minimize_phi_and_dist" value="true" />
    <param name="init_controls" value="true" />
    <param name="use_th_constraints" value="true" />
    <param name="use_xy_constraints" value="true" />
    <param name="phi_min" value="-1.0" /> <!-- -0.9 -->
    <param name="phi_max" value="1.0" />  <!-- 0.9 -->
    <param name="reassign_constraints" value="true" />
    <param name="reassign_iters" value="1" />
    <param name="reassign_min_distance" value="0.1" />
  </node>  

  <node pkg="orunav_vehicle_execution" type="orunav_vehicle_execution_node" name="orunav_vehicle_execution_node_citi2" output="screen">
    <param name="robot_id" value="2" />

    <param name="visualize" value="true" />
    <param name="max_tracking_error" value="-1." />
    <param name="use_forks" value="true" />

    <param name="max_vel" value="0.3" />
    <param name="max_rotational_vel" value="0.3" />
    <param name="max_acc" value="0.1" />
    <param name="use_update_task_service" value="false" />

    <remap from="controller/reports" to="control/controller/reports" />
    <remap from="controller/commands" to="control/controller/commands" />
    <remap from="fork/report" to="control/fork/report" />
    <remap from="fork/command" to="control/fork_command" />
    <remap from="controller/trajectories" to="control/controller/trajectories" />

  </node>

  </group>

</launch>
